### PR TITLE
Update Swagger reference for FR

### PR DESCRIPTION
**Pull request by maririos**
~Creating as draft so people can see the changes that are coming. 
Ci is going to fail because of https://github.com/Azure/autorest.csharp/issues/652~

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs [hunk 1]
```diff
@@ -93,13 +93,12 @@ public DateTime AsDate()
...
 
-            DateTime date = default;
-            if (!DateTime.TryParse(_fieldValue.ValueDate, out date))
+            if (!_fieldValue.ValueDate.HasValue)
             {
                 throw new InvalidOperationException($"Cannot parse Date value {_fieldValue.ValueDate}.");
             }
 
-            return date;
+            return _fieldValue.ValueDate.Value.UtcDateTime;
```
Comment (maririos) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs:101]: @annelo-msft based on https://github.com/Azure/azure-sdk-for-net/issues/11120#issuecomment-612175895 
is it ok to use `UtcDateTime` here?

Comment (annelo-msft) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs:101]: The return type is correct.  I'm not 100% certain what time zone is returned for receipts.  Have you validated that the time is the same as the time returned for the receipt?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs [hunk 2]
```diff
@@ -178,7 +178,7 @@ public virtual Pageable<CustomFormModelInfo> GetModelInfos(CancellationToken can
         [ForwardsClientCalls]
         public virtual AsyncPageable<CustomFormModelInfo> GetModelInfosAsync(CancellationToken cancellationToken = default)
         {
-            return ServiceClient.GetCustomModelsPageableModelInfoAsync(GetModelOptions.Full, cancellationToken);
+            return ServiceClient.GetCustomModelsPageableModelInfoAsync(cancellationToken);
```
Comment (annelo-msft) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs:181]: Are Full/Summary not needed now?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/FieldValue_internal.Serialization.cs [hunk 3]
```diff
@@ -17,7 +18,7 @@ internal static FieldValue_internal DeserializeFieldValue_internal(JsonElement e
         {
             FieldValueType type = default;
             string valueString = default;
-            string valueDate = default;
+            DateTimeOffset? valueDate = default;
             string valueTime = default;
```
Comment (annelo-msft) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/FieldValue_internal.Serialization.cs:22]: Is Time support in now as well?

**Review by kinelski — APPROVED** [review]
LGTM!

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs (hunk 2 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs:181]: No. Now we have different ways to do it. Either by calling `List...` or `Get...` 
See https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json#L457

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/FieldValue_internal.Serialization.cs (hunk 3 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/FieldValue_internal.Serialization.cs:22]: It is :) 
although the changes to the swagger are still in PR. See https://github.com/Azure/azure-rest-api-specs/pull/8977

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs (hunk 1 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs:101]: Yes. so far no different

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs (hunk 2 above)
Comment (annelo-msft) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs:181]: Oh I see how the swagger changed to generate this differently.  I believe where we had Full, we'll want ListCustomModels and where we had Summary, we'll want GetCustomModels based on the new Swagger.

**Review by annelo-msft — APPROVED** [review]

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs (hunk 2 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs:181]: Correct

**Pull request merged**